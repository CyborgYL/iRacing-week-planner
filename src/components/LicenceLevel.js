// @flow

import * as React from 'react';
import classnames from 'classnames';

import levelToClass from '../lib/levelToClass';
import styles from './styles/licenceLevel.module.scss';

type Props = {
  licence: number,
  effective?: boolean,
};

export default function LicenceLevel({ licence, effective = false }: Props): React.Node {
  return (
    <div
      className={classnames(
        styles.licenceLevelComponent,
        styles[`licence${levelToClass(licence, effective).toUpperCase()}`],
      )}
    >
      <span className="licenceLetter">{levelToClass(licence, effective)}</span>
      <span className="licenceText">
        {effective ? null : (((licence - 1) % 4) + 1).toFixed(2)}
      </span>
    </div>
  );
}
