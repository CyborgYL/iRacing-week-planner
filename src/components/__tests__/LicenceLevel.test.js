import React from 'react';
import { shallow } from 'enzyme';
import { describe, expect, test } from '@jest/globals';

import LicenceLevel from '../LicenceLevel';

describe('components/LicenceLevel', () => {
  test('renders for high D Class', () => {
    const render = shallow(<LicenceLevel licence={8} />);
    expect(render).toMatchSnapshot();

    expect(render.find('.licenceLetter').text()).toBe('D');
    expect(render.find('.licenceText').text()).toBe('4.00');
    expect(render.hasClass('licenceD')).toBe(true);
  });

  test('renders for Pro', () => {
    const render = shallow(<LicenceLevel licence={21} />);
    expect(render).toMatchSnapshot();

    expect(render.find('.licenceText').text()).toBe('1.00');
    expect(render.hasClass('licenceP')).toBe(true);
  });

  test('renders for float levels', () => {
    const render = shallow(<LicenceLevel licence={7.5} />);
    expect(render).toMatchSnapshot();

    expect(render.find('.licenceLetter').text()).toBe('D');
    expect(render.find('.licenceText').text()).toBe('3.50');
    expect(render.hasClass('licenceD')).toBe(true);
  });

  test('renders for other float levels', () => {
    const render = shallow(<LicenceLevel licence={8.5} />);
    expect(render).toMatchSnapshot();

    expect(render.find('.licenceLetter').text()).toBe('D');
    expect(render.find('.licenceText').text()).toBe('4.50');
    expect(render.hasClass('licenceD')).toBe(true);
  });

  test('renders effective licence', () => {
    const render = shallow(<LicenceLevel licence={8} effective />);
    expect(render).toMatchSnapshot();

    expect(render.find('.licenceLetter').text()).toBe('C');
    expect(render.find('.licenceText').text()).toBe('');
    expect(render.hasClass('licenceC')).toBe(true);
  });
});
