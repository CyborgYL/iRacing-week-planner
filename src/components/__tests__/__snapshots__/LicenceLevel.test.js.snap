// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/LicenceLevel renders effective licence 1`] = `
<div
  className="licenceLevelComponent licenceC"
>
  <span
    className="licenceLetter"
  >
    C
  </span>
  <span
    className="licenceText"
  />
</div>
`;

exports[`components/LicenceLevel renders for Pro 1`] = `
<div
  className="licenceLevelComponent licenceP"
>
  <span
    className="licenceLetter"
  >
    P
  </span>
  <span
    className="licenceText"
  >
    1.00
  </span>
</div>
`;

exports[`components/LicenceLevel renders for float levels 1`] = `
<div
  className="licenceLevelComponent licenceD"
>
  <span
    className="licenceLetter"
  >
    D
  </span>
  <span
    className="licenceText"
  >
    3.50
  </span>
</div>
`;

exports[`components/LicenceLevel renders for high D Class 1`] = `
<div
  className="licenceLevelComponent licenceD"
>
  <span
    className="licenceLetter"
  >
    D
  </span>
  <span
    className="licenceText"
  >
    4.00
  </span>
</div>
`;

exports[`components/LicenceLevel renders for other float levels 1`] = `
<div
  className="licenceLevelComponent licenceD"
>
  <span
    className="licenceLetter"
  >
    D
  </span>
  <span
    className="licenceText"
  >
    4.50
  </span>
</div>
`;
